### PR TITLE
Some cancel telemetry

### PR DESCRIPTION
**Pull request by StephenWeatherford**
The only things I can put in quickly, but should handle the major scenarios.

### CODE REVIEW

File: src/explorer/SiteTreeItem.ts [hunk 1]
```diff
@@ -110,9 +110,10 @@ export abstract class SiteTreeItem implements IAzureParentTreeItem {
         outputChannel: OutputChannel,
         telemetryReporter: TelemetryReporter,
         configurationSectionName: string,
-        confirmDeployment: boolean = true
+        confirmDeployment: boolean = true,
+        telemetryProperties: { [key: string]: string }
```
Comment (EricJizbaMSFT) [inline src/explorer/SiteTreeItem.ts]: `TelemetryProperties`

File: src/extension.ts [hunk 2]
```diff
@@ -153,16 +157,23 @@ export function activate(context: vscode.ExtensionContext): void {
...
+            fsPath = await util.showWorkspaceFoldersQuickPick("Select the folder to deploy", this.properties);
             node = target;
         }
 
         if (!node) {
-            node = <IAzureNode<WebAppTreeItem>>await tree.showNodePicker(WebAppTreeItem.contextValue);
+            try {
+                node = <IAzureNode<WebAppTreeItem>>await tree.showNodePicker(WebAppTreeItem.contextValue);
+            } catch (err2) {
+                if (err2 instanceof UserCancelledError) {
```
Comment (EricJizbaMSFT) [inline src/extension.ts]: Do `parseError(err2).isUserCancelledError` (we've had weird bugs in the past where `instanceof` doesn't work if there are somehow multiple versions of the ui package)

File: src/extension.ts [hunk 3]
```diff
@@ -206,9 +217,9 @@ export function activate(context: vscode.ExtensionContext): void {
         const yesButton: vscode.MessageItem = { title: 'Yes' };
         const noButton: vscode.MessageItem = { title: 'No', isCloseAffordance: true };
         if (await vscode.window.showInformationMessage('Deploy to deployment slot?', yesButton, noButton) === yesButton) {
```
Comment (EricJizbaMSFT) [inline src/extension.ts]: Suggest doing the same 'deployingToWebApp' that you did for CreateWebApp above

File: src/util.ts [hunk 4]
```diff
@@ -87,6 +87,7 @@ export async function showWorkspaceFoldersQuickPick(placeHolderString: string):
     const pickedItem = await vscode.window.showQuickPick(folderQuickPickItems, folderQuickPickOption);
 
     if (!pickedItem) {
+        telemetryProperties.cancelStep = placeHolderString;
```
Comment (EricJizbaMSFT) [inline src/util.ts]: nit: I'd prefer if this was just a constant string like "showWorkspaceFolders". We're never going to have _multiple_ calls to showWorkspaceFolders in one command so I'd rather have something consistent across all commands

File: src/util.ts [hunk 5]
```diff
@@ -99,6 +100,7 @@ export async function showWorkspaceFoldersQuickPick(placeHolderString: string):
         });
 
         if (!browseResult) {
+            telemetryProperties.cancelStep = 'showWorkspaceFoldersQuickPick';
```
Comment (EricJizbaMSFT) [inline src/util.ts]: How about 'showWorkspaceFoldersBrowse'

Comment (StephenWeatherford) [reply src/util.ts]: Fixed all.

**Pull request merged**